cmd/paths.go

```diff
@@ -256,77 +256,124 @@ func pathsFetchCmd(a *appState) *cobra.Command {
 $ %s paths fetch --home %s
 $ %s pth fch`, appName, defaultHome, appName)),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// Clone the GH repo to tmp dir, we will extract the path files from here
-			localRepo, err := os.MkdirTemp("", "")
-			if err != nil {
-				return err
+			chains := []string{}
+			for chain_name := range a.Config.Chains {
+				chains = append(chains, chain_name)
 			}
 
-			if _, err = git.PlainClone(localRepo, false, &git.CloneOptions{
-				URL:           REPOURL,
-				Progress:      io.Discard,
-				ReferenceName: "refs/heads/main",
-			}); err != nil {
-				return err
-			}
+			combinations := permutations(chains)
 
-			// Try to fetch path info for each configured chain that has canonical chain/path info in the GH repo
-			for _, srcChain := range a.Config.Chains {
-				for _, dstChain := range a.Config.Chains {
+			client := github.NewClient(nil)
+			gh_ctx := context.Background()
```

Comment:
Instead of `context.Background()`, this should use the context associated with `cmd`, at `cmd.Context()`. Its use can be inlined in the one call, no need for this variable.